### ISSUE TITLE
Check for allowed regions

### ISSUE THREAD

**Issue by phackstock** [IAMconsortium/nomenclature#26]
Building on #22, the next step should be to integrate `RegionAggregationMapping` with `DataStructureDefinition` in order to verify that all specified native regions as well as common regions are allowed.

**Comment by danielhuppmann** [comment]
Happy to share my thoughts...

1. There is a misunderstanding about the aim of the `DataStructureDefinition.validate` method - it validates an **IamDataFrame** instance (e.g., an upload by a modelling team) against an instance of the **DataStructureDefinition** initialized from a project-specific directory (i.e., correct variables and regions). It is *not* a validation of the nomenclature-yaml files. This (implicitly) already occurs when initializing a **DataStructureDefinition** instance from a directory.

2. The guiding question should be: how to execute the region-aggregation? One option that I see is the following:

   ```python
   dsd = nomenclature.DataStructureDefinition("<path/to/definitions>")
   reg = nomenclature.RegionProcessor("<path/to/definitions>", dsd)

   df = pyam.IamDataFrame(df)d
   dsd.validate(df)
   reg.apply(df)
   ```

   The validation that all native[-renamed] and common regions are defined in the **DataStructureDefinition** instance should happen as part of the initialization of the **RegionProcessor** instance.

   Alternatively, the collection of region-aggregation-mappings could be implemented as a module of the **DataStructureDefinition**.

3. When do we check the validity of configuration files and mappings? This should happen in every (project-specific) repository that has such files - executed via GitHub Actions upon a PR or push, so that invalid or inconsistent files are unlikely to end up on the main branch of any repo.

   The **nomenclature** package already has a testing module ([see here](https://github.com/IAMconsortium/nomenclature/blob/main/nomenclature/testing.py)), which is currently used as part of the automated testing of the irp-internal-workflow repository ([see here](https://github.com/iiasa/irp-internal-workflow/blob/main/tests/test_core.py)). This should be further developed, I'll start an issue.

   Furthermore, the way the code is structured now, all region-mapping files have to be parsed anyway to find the "correct" mapping file when importing the package.

   In short, we want to do 1 and we have to do 2 (unless we restructure the code and have a central "directory" with the mapping of model-to-mapping-file).
